docs/pbc/all/payment-service-provider/202410.0/base-shop/hosted-payment-page-back-button-handling.md

```diff
@@ -0,0 +1,117 @@
+template: concept-topic-template
+---
+
+When integrating payment service providers (PSPs) that use hosted payment pages, the application needs to handle browser navigation to maintaining order consistency and stock management. This document explains how to configure applications with hosted payment pages to handle the browser back button action.
+
+## Order persistence issue
+
+When a customer is redirected from the Spryker checkout summary page to an external PSP's hosted payment page, using the back button creates an order persistence issue. The order is already created in the system with a unique OrderReference, preventing the customer from completing the purchase if they attempt to proceed again through checkout.
+
+This issue is particularly critical in the following cases:
+* The last item in stock was part of the order
+* The customer wants to change their order after being redirected to the payment page
+* The order reference is already generated and persisted
+
+## Solution
+
+Implement a mechanism to detect and handle cases where customers return from a hosted payment page using the back button. In such cases, the system cancels a previously created order and lets the customer to proceed with a new order.
+
+## Implement back button handling
+
+1. To support browser back button behavior for hosted payment pages, register the `PaymentAppCancelOrderOnSummaryPageAfterRedirectFromHostedPaymentPagePlugin` in your project's `CheckoutPageDependencyProvider`.
+
+Add the following methods to `src/Pyz/Yves/CheckoutPage/CheckoutPageDependencyProvider.php`:
+
+```php
+<?php
+
+namespace Pyz\Yves\CheckoutPage;
+
+use SprykerShop\Yves\PaymentAppWidget\Plugin\CheckoutPage\PaymentAppCancelOrderOnSummaryPageAfterRedirectFromHostedPaymentPagePlugin;
+
+class CheckoutPageDependencyProvider extends SprykerCheckoutPageDependencyProvider
+{
+    /**
+     * @return array<\SprykerShop\Yves\CheckoutPageExtension\Dependency\Plugin\CheckoutStepPreConditionPluginInterface>
+     */
+    protected function getCheckoutSummaryStepPreConditionPlugins(): array
+    {
+        return [
+            new PaymentAppCancelOrderOnSummaryPageAfterRedirectFromHostedPaymentPagePlugin(),
+        ];
+    }
+
+    /**
+     * @return array<\SprykerShop\Yves\CheckoutPageExtension\Dependency\Plugin\CheckoutStepPostConditionPluginInterface>
+     */
+    protected function getCheckoutSummaryStepPostConditionPlugins(): array
+    {
+        return [
+            new PaymentAppCancelOrderOnSummaryPageAfterRedirectFromHostedPaymentPagePlugin(),
+        ];
+    }
+}
+```
+
+2. Update your OMS state machine configuration by adding the `exclude from customer` flag to the `payment cancellation pending` state:
+
+
+**SalesPayment/config/Zed/Oms/Subprocess/PaymentCancel01.xml**
+```xml
+<state name="payment cancellation pending" display="oms.state.reservation-cancellation-pending">
+    <flag>exclude from customer</flag>
+</state>
+```
+
+{% info_block warningBox "" %}
```

Comment:
Why is this highlighted as a warning when the reader has to do it anyway? It's part of the guide

Reply:
It was meant to be very important to not be missed.

Updated.